test/functional/ui/popupmenu_spec.lua

```diff
@@ -1790,6 +1790,36 @@ describe('builtin popupmenu', function()
       end)
     end
 
+    it('pum popup preview', function()
+      screen:set_default_attr_ids({})
+      screen:try_resize(30, 8)
+      exec([[
+        funct Omni_test(findstart, base)
+          if a:findstart
+            return col(".") - 1
+          endif
+          return [#{word: "one", info: "1info"}, #{word: "two", info: "2info"}, #{word: "three"}]
+        endfunc
+        set omnifunc=Omni_test
+        set completeopt=menu,popup
+        funct Set_info()
+          let comp_info = complete_info()
+          if has_key(comp_info, 'preview_bufnr')
+            call nvim_set_option_value('filetype', 'markdown', {'buf': comp_info['preview_bufnr']})
+          endif
+          if has_key(v:event, 'selected') && v:event["selected"] == 2
```

Comment:
`v:event["selected"] == 2` this makes the info apply to the third item (zero-indexed), though in the test we get to the first item only.

Reply:
It simulates asynchronously setting info after `feed<C-n>` twice. three is selected then.